web/src/engine/main/src/engineConfiguration.ts

```diff
@@ -0,0 +1,108 @@
+import { DeviceSpec, ManagedPromise, physicalKeyDeviceAlias, SpacebarText } from "@keymanapp/keyboard-processor";
+import { PathConfiguration, PathOptionDefaults, PathOptionSpec } from "keyman/engine/configuration";
+import { Device } from "keyman/engine/device-detect";
+import { KeyboardStub } from "keyman/engine/keyboard-cache";
+
+export class EngineConfiguration {
+  // The app/webview path replaces this during init, but we expect to have something set for this
+  // during engine construction, which occurs earlier.  So no `readonly`, sadly.
+  //
+  // May also be manipulated by Developer's debug-host?
+  public hostDevice: DeviceSpec;
+  readonly sourcePath: string;
+  readonly deferForInitialization: ManagedPromise<void>;
+
+  private _paths: PathConfiguration;
+  private _activateFirstKeyboard: boolean;
+  private _defaultSpacebarText: SpacebarText;
+  private _stubNamespacer?: (KeyboardStub) => void;
+
+  public applyCacheBusting: boolean = false;
+
+  // sourcePath:  see `var sPath =` in kmwbase.ts.  It is not obtainable headlessly.
+  constructor(sourcePath: string, device?: DeviceSpec) {
+    if(!device) {
+      const deviceDetector = new Device();
+      deviceDetector.detect();
+
+      device = deviceDetector.coreSpec;
+    }
+
+    this.sourcePath = sourcePath;
+    this.hostDevice = device;
+    this.deferForInitialization = new ManagedPromise<void>();
+  }
+
+  initialize(options: Required<InitOptionSpec>) {
+    this._paths = new PathConfiguration(options, this.sourcePath);
+    if(typeof options.setActiveOnRegister == 'boolean') {
+      this._activateFirstKeyboard = options.setActiveOnRegister;
+    } else if (typeof options.setActiveOnRegister == 'string') {
+      let str = options.setActiveOnRegister.toLowerCase();
+      this._activateFirstKeyboard = str === 'true';
```

Comment:
Why is this option different types? Seems weird.

Reply:
The documentation at https://help.keyman.com/DEVELOPER/ENGINE/WEB/15.0/reference/core/init does suggest we could just go with `string`.

That said, from `master`: https://github.com/keymanapp/keyman/blob/ecd97d2f2a338b986fad0fa3d3e4a17c42a39116/web/src/engine/main/kmwbase.ts#L94-L102

I suppose that TODO can be deferred instead.

Reply:
https://github.com/search?q=org%3Akeymanapp%20setActiveOnRegister&type=code

Yuck! This is the type of thing where the string type should be marked as deprecated in 17.0 and replaced with a boolean in 18.0+. For now though, it is _only_ documented as supporting string.

Reply:
So given KM Developer Server passes a boolean, and keymanweb.com passes a string, I'd be really happy to just turn this into Boolean and mark it as a breaking change in the release notes, fixing keymanweb.com at the same time. It's a super simple fix for anyone using kmw on their own site, and they'll probably feel good about themselves when they convert `"true"` into `true` too!

Reply:
See #4304 for history of this

Reply:
> fixing keymanweb.com at the same time

I'm not seeing a super-simple way to make an update for this right now for keymanweb.com without including lines 40-41 above.  I suppose we'd need to make that updates come `beta` instead?

I've gone ahead and removed those lines in my most recent commit.

Reply:
For keymanweb.com we should do that when this merges into master, and then version-compare to determine which data type to use.

Reply:
Added as a TODO item on #8047.